### PR TITLE
Create csv_to_sqlite.R

### PR DESCRIPTION
**Pull request by damianooldoni**
I found this function in INBO tutorial [large-files-R.Rmd](https://github.com/inbo/tutorials/blob/master/source/data-handling/large-files-R.Rmd). I would like to use this function in TrIAS pipeline. In general, it would be nice to have it in package `inborutils` for being shared with everyone.

### CODE REVIEW

**Comment by damianooldoni** [comment]
I just noticed I forgot to add @export and @importFrom parameters in documentation. Wait for review, please.

File: R/csv_to_sqlite.R [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+
+    # read an extract of the data to extract the colnames and types
+    # to figure out the date and the datetime columns
+    df <- read_delim(csv_file, delim = delim, quote = quote,
+                     escape_backslash = escape_backslash,
+                     col_names = col_names, col_types = col_types,
+                     locale = locale, na = na, quoted_na = quoted_na,
+                     comment = comment, trim_ws = trim_ws,
+                     skip = skip, escape_double = escape_double,
+                     n_max = pre_process_size)
```
Comment (stijnvanhoey) [inline R/csv_to_sqlite.R]: Would it not be more convenient to use the [three dot ellipsis](https://www.r-bloggers.com/r-three-dots-ellipsis/) to pass the *additional* arguments (e.g. `delim`, `col_names`, `quote` directly to `the read_delim` function? That would for example also not require function adaptations in this function when `read_delim` get changed more upstream. Moreover, would release the function from a lot of the arguments, making it clearer.

**Review by stijnvanhoey — COMMENTED** [review]
Thanks for converting this to the inborutils package! Just a single comment from my side...

File: R/csv_to_sqlite.R (hunk 1 above)
Comment (damianooldoni) [reply R/csv_to_sqlite.R]: Ah yes, interesting! I will test it and let you know.

Comment (damianooldoni) [reply R/csv_to_sqlite.R]: As promised, done ffe889806bd9c647b5ff83483af4bae426f71eba

**Comment by damianooldoni** [comment]
Thanks for suggesting the use of ellipsis! Code is much more readable and nice. I tested the function by converting the csv file `2016-04-20-processed-logs-big-file-example.csv` as shown in tutorial [Handling large files in R](https://github.com/inbo/tutorials/blob/master/source/data-handling/large-files-R.Rmd).

Convert csv to database:

```r
sqlite_file <- "example2.sqlite"
table_name <- "birdtracks"
inborutils::csv_to_sqlite(csv_file = paste0("../data/interim/", csv.name),
              sqlite_file = paste0("../data/interim/", sqlite_file), 
              table_name =table_name,
              trim_ws = TRUE, escape_double = FALSE)
```

Results got from the converted database using `inborutils::csv_to_file()`:

```r
my_db <- src_sqlite(paste0("../data/interim/", sqlite_file), create = FALSE)
bird_tracking <- tbl(my_db, "birdtracks")
results <- bird_tracking %>%
  filter(device_info_serial == 860) %>%
  select(date_time, latitude, longitude, altitude) %>%
  filter(date_time < "2014-07-01") %>%
  filter(date_time > "2014-03-01") %>%
  as_tibble()
head(results)

  date_time           latitude longitude altitude
  <chr>                  <dbl>     <dbl>    <int>
1 2014-03-10 12:43:37     44.0     -7.59      626
2 2014-03-10 12:58:32     44.1     -7.67      405
3 2014-03-10 13:13:52     44.1     -7.69      326
4 2014-03-10 13:28:57     44.1     -7.71      250
5 2014-03-10 13:43:54     44.1     -7.72      174
6 2014-03-10 13:59:06     44.1     -7.73       23
```

Results got from the downloaded database `2016-04-20-processed-logs-big-file-example.db`:

```r
my_db_test <- src_sqlite(paste0("../data/interim/", db.name), create = FALSE)
bird_tracking_test <- tbl(my_db, "birdtracks")
results_test <- bird_tracking_test %>%
  filter(device_info_serial == 860) %>%
  select(date_time, latitude, longitude, altitude) %>%
  filter(date_time < "2014-07-01") %>%
  filter(date_time > "2014-03-01") %>%
  as_tibble()
head(results_test)

  date_time           latitude longitude altitude
  <chr>                  <dbl>     <dbl>    <int>
1 2014-03-10 12:43:37     44.0     -7.59      626
2 2014-03-10 12:58:32     44.1     -7.67      405
3 2014-03-10 13:13:52     44.1     -7.69      326
4 2014-03-10 13:28:57     44.1     -7.71      250
5 2014-03-10 13:43:54     44.1     -7.72      174
6 2014-03-10 13:59:06     44.1     -7.73       23
```

`results`and `results_test` are identical as expected:

```r
> ncol(results) == ncol(results_test)
[1] TRUE
> nrow(results) == nrow(results_test)
[1] TRUE
> all(results == results_test)
[1] TRUE
```

@stijnvanhoey : merge if you agree.

**Comment by damianooldoni** [comment]
Ah, last idea, striving for perfection: maybe should I add an example in documentation too? Maybe the same example given in tutorial and in the previous comment? I think it will make the documentation better.

**Comment by stijnvanhoey** [comment]
Good idea, maybe add an example within the roxygen header:

```
#'@examples
#'\dontrun{... }
```

And refer to the complete tutorial page of the tutorials website?

**Comment by damianooldoni** [comment]
Yes, that is what I had in mind! The typical `@examples{}` section.
About link to tutorial: great idea. Thanks a lot. I do it immediately.

**Comment by damianooldoni** [comment]
Done https://github.com/inbo/inborutils/pull/32/commits/981d07e4b83ae61528af312f171a0a3abcb79021

**Comment by stijnvanhoey** [comment]
Cool! Nice work

**Pull request merged**